File: src/dotnet/UnitTests/Day1Tests.cs
using Domain.Day1;
using FluentAssertions;

namespace UnitTests;

public class Day1Tests
{
    private readonly string[] _calories;

    public Day1Tests()
    {
        _calories = File.ReadAllText(@"./TestData/Day1_Data.txt")
            .Split("\n\n");
    }

    [Fact]
    public void GetsMaxCalories()
    {
        // Arrange
        const int expectedResult = 24000;

        // Act
        var result = Day1.GetsMaxCalories(_calories);

        // Assess
        result.Should().Be(expectedResult);
    }

    [Fact]
    public void GetsSumOfTopThree()
    {
        // Arrange
        const int expectedResult = 45000;

        // Act
        var result = Day1.GetsSumOfTopThree(_calories);

        // Assess
        result.Should().Be(expectedResult);
    }
}

File: src/python/unit_tests/day1_tests.py
import unittest
from pathlib import Path
from domain.day1.day1 import Day1


class TestDay1(unittest.TestCase):

    data = Path('test_data/Day1_data.txt').read_text()

    def test_top3_calories(self):
        # arrange
        expectedResult = 45000

        # act
        calories = self.data.split("\n\n")
        result = Day1.get_top3_calories(calories)

        # assert
        self.assertEqual(expectedResult, result)

    def test_highest_calories(self):
        # arrange
        expectedResult = 24000

        # act
        calories = self.data.split("\n\n")
        result = Day1.get_highest_calories(calories)

        # assert
        self.assertEqual(expectedResult, result)


if __name__ == '__main__':
    unittest.main()
